sniffles/nfa.py

```diff
@@ -427,10 +434,13 @@ def op_char(self, sp):
 
     def op_circ(self, sp):
         self.cp += 1
-        fallback = self.nfa.start
-        self.nfa.start = get_nfa_state()
-        self.nfa.start.add_tx(E, fallback)
-        return self.nfa.start
+        # it can cause a problem with
+        # multiple regular expression
+        # fallback = self.nfa.start
+        # self.nfa.start = get_nfa_state()
```

Comment:
Go ahead and delete these lines.  If we need a history I can just go back through the revision history.